packages/scene-composer/src/augmentations/components/three-fiber/viewpoint/ViewCursorWidget.tsx

```diff
@@ -1,76 +1,94 @@
-import * as THREE from 'three';
-import React, { useContext, useEffect, useMemo, useRef } from 'react';
-import { extend, useLoader, useThree } from '@react-three/fiber';
-import { SVGLoader, SVGResult } from 'three-stdlib';
+import React, { useCallback, useContext, useMemo, useRef } from 'react';
+import { useFrame, useLoader, useThree } from '@react-three/fiber';
+import { AlwaysDepth, Box3, Color, DoubleSide, Group, Mesh, MeshBasicMaterial, Object3D, ShapeGeometry, Vector3 } from 'three';
+import { SVGLoader } from 'three-stdlib';
 
-import { ViewCursorMoveIcon, ViewCursorEditIcon } from '../../../../assets';
-import svgIconToWidgetVisual from '../common/SvgIconToWidgetVisual';
-import { WidgetVisual } from '../../../three/visuals';
-import { ViewCursor, Viewpoint, ViewpointState } from '../../../three';
+import { ViewCursorEditIcon, ViewCursorMoveIcon } from '../../../../assets';
+import { getIntersectionTransform } from '../../../../utils/raycastUtils';
 import { sceneComposerIdContext } from '../../../../common/sceneComposerIdContext';
 import { useEditorState } from '../../../../store';
 
-// Adds the custom objects to React Three Fiber
-extend({ ViewCursor, Viewpoint, WidgetVisual });
-
-const AsyncLoadViewCursorWidget: React.FC = () => {
+export const ViewCursorWidget = () => {
+  const { scene } = useThree();
+  const ref = useRef<Object3D>(null);
   const sceneComposerId = useContext(sceneComposerIdContext);
-  const { cursorPosition, cursorLookAt, cursorVisible, cursorStyle } = useEditorState(sceneComposerId);
-  const viewCursorRef = useRef<Viewpoint>(null);
-  const { camera } = useThree();
+  const { cursorVisible, cursorStyle } = useEditorState(sceneComposerId);
 
-  const moveVisual = useMemo(() => {
-    const svgData: SVGResult = useLoader(SVGLoader, ViewCursorMoveIcon.dataUri);
-    return svgIconToWidgetVisual(svgData, ViewpointState.Deselected, false, { userData: { isCursor: true } });
-  }, []);
+  const svg = cursorStyle === 'edit' ? ViewCursorEditIcon : ViewCursorMoveIcon;
+  const data = useLoader(SVGLoader, svg.dataUri);
 
-  const editVisual = useMemo(() => {
-    const svgData: SVGResult = useLoader(SVGLoader, ViewCursorEditIcon.dataUri);
-    return svgIconToWidgetVisual(svgData, ViewpointState.Selected, false, { userData: { isCursor: true } });
-  }, []);
+  /* istanbul ignore next */
```

Comment:
Are we not testing this part because it's hard, or because it's unnecessary? If it's hard, and we don't have time for it, I'd prefer we actually just drop coverage rather than creating a false sense of security.

Reply:
`resetObjectCenter` is used to keep an `svg` centered to the mouse pointer so it would be more appropriately tested with end-to-end tests. Unit test coverage for this would be useless IMO. However, if you don't agree we can discuss. 

The point you made below about moving the `shape` rendering into utils makes sense & would be more testable in that aspect.

Reply:
`svg` testing is included in upcoming push.